string name
int32 lower
int32 upper
---
int32 error_code
string error_description